routing/pathfind.go

```diff
@@ -497,7 +499,8 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		// If there is only not enough capacity on a single route, it
 		// may still be possible to complete the payment by splitting.
 		if max < amt {
-			return nil, errNoPathFound
+			splitNeeded = true
+			amt = max
```

Comment:
Does this work ok when routing fees to be paid on top of `amt` are factored in too, or will trying to deliver `max` to the final hop result in the need to pay `max + fees` on the local channel?